benchmark.py

```
@@ -0,0 +1,833 @@
+        
+        print(f"  True Positives (expected match, got match): {total_tp:.1f}")
+        print(f"  True Negatives (expected miss, got miss): {total_tn:.1f}")
+        print(f"  False Positives (expected miss, got match): {total_fp:.1f}")
+        print(f"  False Negatives (expected match, got miss): {total_fn:.1f}")
+        
+        if total_fp > 0:
+            print(f"  ⚠️  Algorithm found unexpected matches ({total_fp:.1f} false positives)")
+        if total_fn > 0:
+            print(f"  ⚠️  Algorithm missed expected matches ({total_fn:.1f} false negatives)")
+        if total_tp == 0 and total_fp == 0:
+            print(f"  ℹ️   No matches found at all - algorithm may be very strict or incompatible data format")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)